docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,403 @@
+## simulateTransaction Examples and Tutorials Guide
+
+### Table of Contents
+
+- [simulateTransaction Examples and Tutorials Guide](#simulatetransaction-examples-and-tutorials-guide)
+  - [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [When to Use `simulateTransaction` (whenever you use soroban :D )](#when-to-use-simulatetransaction-whenever-you-use-soroban-d-)
+- [How to Call `simulateTransaction`](#how-to-call-simulatetransaction)
+  - [Using Fetch](#using-fetch)
+  - [Using the JavaScript SDK](#using-the-javascript-sdk)
+- [Understanding the Footprint](#understanding-the-footprint)
+- [Assembling a Transaction](#assembling-a-transaction)
+  - [javascript-sdk-assembleTransaction](#javascript-sdk-assembletransaction)
+- [Handling Archived Ledger Entries](#handling-archived-ledger-entries)
+- [Fees and Resource Usage](#fees-and-resource-usage)
+- [Preflight Error Handling](#preflight-error-handling)
+  - [Common Preflight Errors](#common-preflight-errors)
+- [Backend Code and Workflow](#backend-code-and-workflow)
+- [Further Reading](#further-reading)
+
+## Overview
+
+The `simulateTransaction` endpoint in Soroban RPC allows you to submit a trial contract invocation to simulate how it would be executed by the Stellar network. This simulation calculates the effective transaction data, required authorizations, and minimal resource fee. It provides a way to test and analyze the potential outcomes of a transaction without actually submitting it to the network. It can be a nice way to get contract data as well sometimes.
+
+While calling the method on the rpc server is not the ONLY way to simulate a transaction, it will likely be the most common and easiest way.
+
+Here we will look at the objects involved and their definitions.
+
+## RPC Servers
+
+Stellar Development Foundation provides a testnet RPC endpoint at For Public network providers please refer to the [Ecosystem RPC Providers] list. At the time of writing this documentation there are two public and free RPC providers, and many more you can subscribe to.
```

Comment:
I had added them because it's a bit hard to find a direct link to one from the ecosystem provider list but we can leave that out.  I will go through your other changes later today also.